#!/bin/bash

# Create the header of the table
echo -e "contig\tnumber_of_samples\tnumber_of_records\tnumber_of_noalts\tnumber_of_snps\tnumber_of_mnps\tnumber_of_indels\tnumber_of_others\tnumber_of_multiallelic_sites\tnumber_of_multiallelic_snps" > summary_table_filtered.txt

# Loop through each file
for file in *.txt; do
  contig=$(basename "$file" .txt)
  number_of_samples=$(grep -i "number of samples" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')
  number_of_records=$(grep -i "number of records" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')
  number_of_noalts=$(grep -i "number of no-ALTs" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')
  number_of_snps=$(grep -i "number of SNPs" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')
  number_of_mnps=$(grep -i "number of MNPs" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')
  number_of_indels=$(grep -i "number of indels" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')
  number_of_others=$(grep -i "number of others" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')
  number_of_multiallelic_sites=$(grep -i "number of multiallelic sites" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')
  number_of_multiallelic_snps=$(grep -i "number of multiallelic SNP sites" "$file" | grep -v "^#" | cut -d ":" -f 2 | tr -d '[:space:]')

  echo -e "$contig\t$number_of_samples\t$number_of_records\t$number_of_noalts\t$number_of_snps\t$number_of_mnps\t$number_of_indels\t$number_of_others\t$number_of_multiallelic_sites\t$number_of_multiallelic_snps" >> summary_table_filtered.txt
done

echo "Filtered summary table generated: summary_table_filtered.txt"

#Explanation:
#Initializes the summary file with a header.

#Loops over all .txt files generated by bcftools stats.

#Extracts relevant statistics using grep and cut.

#Appends results in tab-delimited format, filters out commented lines.

#Creates a new summary table with only relevant data.
